### PR TITLE
File IO Stdlib functions

### PR DESCRIPTION
**Pull request by ishaankolluri**
Putting out the PR so everyone can see the implementation. Let me know if anything can be improved. 

Things to potentially add to this PR:
* `readLine`?
* Graph Plotting Library to `lib.c` (this could go in a different PR for atomicity).

### CODE REVIEW

File: src/stdlib/lib.c [hunk 1]
```diff
@@ -104,177 +137,170 @@ value_p printd(subrange_p whatever, subrange_p text) {
...
+	double val = floor(initial);
+	return new_number(val);
+}
+
+value_p extend_open(subrange_p range_one, subrange_p range_two){
+	FILE *val;
+	if(!assertSingle(range_one) || !assertSingle(range_two) || open_num_files + 1 > MAX_FILES) return new_val();
+	value_p filename = get_val(range_one, 0, 0);
+	value_p mode = get_val(range_two, 0,0);
+	val = fopen(filename->str->text, mode->str->text);
```
Comment (Neitsch) [inline src/stdlib/lib.c:268]: Assert string for filename and mode, or write assertSingleString

File: src/stdlib/lib.c [hunk 2]
```diff
@@ -104,177 +137,170 @@ value_p printd(subrange_p whatever, subrange_p text) {
...
+
+value_p extend_close(subrange_p range){
+	if(!assertSingleNumber(range)) {
+		// Per the LRM this is actually supposed to crash the program.
+		fprintf(stderr, "EXITING - Attempted to close something that was not a valid file pointer\n");
+		exit(-1);
+	}
+
+	int fileNum = (int) get_number(range);
+	if (fileNum > open_num_files || open_files[fileNum] == NULL) {
```
Comment (Neitsch) [inline src/stdlib/lib.c:283]: I don't see open_num_files ever being decremented, but if it was, this line could cause issues. Specifically use fileNum > MAX_FILES. Consider case file 1 and 2 are opened. 1 is closed, open_num_files is decremented to 1, then we get into the error condition.

File: src/stdlib/lib.c [hunk 3]
```diff
@@ -104,177 +137,170 @@ value_p printd(subrange_p whatever, subrange_p text) {
...
-	value_p result = new_val();
-	setNumeric(result,val);
-	setFlag(result, FLAG_NUMBER);
+	if(!assertSingleNumber(range)) return new_val();
+	double initial = get_number(range);
+	double val = floor(initial);
+	return new_number(val);
+}
+
+value_p extend_open(subrange_p range_one, subrange_p range_two){
```
Comment (Neitsch) [inline src/stdlib/lib.c]: More descriptive parameter names for all methods please

File: src/stdlib/lib.c [hunk 4]
```diff
@@ -104,177 +137,170 @@ value_p printd(subrange_p whatever, subrange_p text) {
...
+	//edge case: how to return the entire contents of the file if n == empty?
+}
+
+value_p extend_write(subrange_p buf, subrange_p f){
+	int val;
+	if(!assertSingle(buf)) return new_val();
+	if(!assertSingle(f)) return new_val();
+	value_p buffer = get_val(buf, 0, 0);
+	value_p fd = get_val(f, 0, 0);
+	val = fwrite(buffer->str->text, 1, sizeof buffer->str->text, open_files[(int)get_number(f)]);
```
Comment (Neitsch) [inline src/stdlib/lib.c]: Assert text/string

**Review by Neitsch — CHANGES_REQUESTED** [review]
Nice

**Comment by oracleofnj** [comment]
Pushed. There are still a couple TODOs in there. FYI I changed argument order so file_handle is always the first argument to read(), write(), and close().

**Review by Neitsch — APPROVED** [review]

**Pull request merged**